### ISSUE TITLE
Game crashing with segfault, unable to find out why.

### ISSUE THREAD

**Issue by Cabanur** [ValveSoftware/Dota-2#1720]
#### Your system information

* System information from steam: [here](https://gist.github.com/Cabanur/be3f08865bdd424ca28a309e2ecfbf61)
* Have you checked for system updates?: Yes
* Are you using the latest stable video driver available for your system? Yes
* Have you verified the game files?: Yes

#### Please describe your issue in as much detail as possible:
I'm seeing game crashes 2-3 times per game. Can't see a pattern as to why they happen.

Here's what my syslog says around the crash:

```
Mar 17 17:44:02 sobremesa crash_20200317173408_1.dmp[22774]: Uploading dump (out-of-process)#012/tmp/dumps/crash_20200317173408_1.dmp
Mar 17 17:44:02 sobremesa kernel: [ 5239.213524] dota2[22008]: segfault at a092c948 ip 00007f2c87d190d9 sp 00007fff885d2f30 error 4 in libclient.so[7f2c85471000+4442000]
Mar 17 17:44:02 sobremesa kernel: [ 5239.213539] Code: 90 55 48 8b bf a0 02 00 00 48 89 e5 5d e9 4f 93 1a 00 90 90 90 90 90 90 90 90 90 90 90 90 90 90 90 55 48 89 e5 53 48 83 ec 08 <48> 8b 9f a0 02 00 00 80 bb 34 01 00 00 00 74 17 48 89 df e8 9f 77
Mar 17 17:44:03 sobremesa crash_20200317173408_1.dmp[22774]: Finished uploading minidump (out-of-process): success = yes
Mar 17 17:44:03 sobremesa crash_20200317173408_1.dmp[22774]: response: Discarded=1
Mar 17 17:44:03 sobremesa crash_20200317173408_1.dmp[22774]: file ''/tmp/dumps/crash_20200317173408_1.dmp'', upload yes: ''Discarded=1''
```

Attatching the crash dumps.
[dumps.zip](https://github.com/ValveSoftware/Dota-2/files/4344745/dumps.zip)

**Comment by Cabanur** [comment]
Attaching more crash dumps in case they're of any use or a pattern can be found. These are under OpenGL.

[dumps.zip](https://github.com/ValveSoftware/Dota-2/files/4351246/dumps.zip)

**Comment by EarthDevour** [comment]
+1 Experiencing same thing for 3-4 months, unable to play at all. Can't rule out my hardware, don't know if that makes sense, but all that started after ram upgrade, OS reinstalled, ran memtest - all OK. In everyday life PC is stable, no crashes no freezes.
If I leave pc with 8GB of ram, it looks like problem is gone. 
If I populate last 2 slots (2x 4GB, 2x 8GB) (24GB total) it starts crashing with segfault error 4 in various libs 
p.s Ram is totally same, only capacity is different.
p.s.s It happens on Vulcan or OpenGL, doesn't matter

**Comment by Cabanur** [comment]
If you can, @EarthDevour, attatch the dumps the game puts in /tmp/dumps when crashing so Valve can check if this is the same issue as mine and maybe find the pattern that leads to the root cause of the issue.

**Comment by EarthDevour** [comment]
[HW Gist](https://gist.githubusercontent.com/EarthDevour/241b6d0d446a2b987e96b21ef8b4faf8/raw/a7165bdc78eb8c6e6d7756417c2983edbfcf243c/gistfile1.txt)

[crash.dmp.zip](https://github.com/ValveSoftware/Dota-2/files/4382729/crash.dmp.zip)

This one on vulcan, but experiencing same on opengl.

`Mar 25 20:21:59 algirdas-MS-7816 kernel: [ 5269.725190] dota2[7546]: segfault at 55e20cf1dcd0 ip 00007ff8289312e6 sp 00007ffec8973db0 error 4 in libscen
esystem.so[7ff8288ca000+3d9000]

Everything is up to date, OS freshly installed yesterday
`

**Comment by ankworld** [comment]
[crash_dota2.zip](https://github.com/ValveSoftware/Dota-2/files/4382954/crash_dota2.zip)

It rarely crashes for me if I play ranked game but it crashes very often on the custom game. And a 100% crash on the Roshpit Champion 4.0 map.

**Comment by ankworld** [comment]
I try with two computers but they have the same result both opengl and vulkan.
1. Laptop with Intel + Nvidia on Solus. (8 GB memory)
2. PC with AMD Ryzen + AMDGPU on Fedora 31. (16 GB memory)

ps. I suggest that you can try with the Roshpit Champion map.

**Comment by EarthDevour** [comment]
My bet is, something is wrong with shared memory. 
When I was searching for any possible cause for this problem, I was surprised how many people has similar issues, from that number its just hard to blame single hardware for it...

**Comment by EarthDevour** [comment]
[Here](https://gist.githubusercontent.com/EarthDevour/2bce2194a2f6ca58a36f29a7a3714238/raw/de3ca4c3b87e4bcc6fef766fbe8121707dac3e84/gdb) is GDB output attached to steam. From start to crash if that helps (and if i'v done it correctly)

**Comment by EarthDevour** [comment]
also, I can see this message in dmesg which is a bit weird:
`[    5.229260] resource sanity check: requesting [mem 0x000c0000-0x000fffff], which spans more than PCI Bus 0000:00 [mem 0x000d0000-0x000d3fff window]
[    5.229361] caller os_map_kernel_space.part.13+0x6d/0x80 [nvidia] mapping multiple BARs
`

**Comment by EarthDevour** [comment]
Dual booted to Windows - same problem. So OS-level things out of discussion.
@Cabanur what vendor Mainboard you have ? I have MSI z87-g43 and Crucial Ballistix RAM(BLS8G3D1609DS1S00)
Obeservations:
2x 4GB ram (same ballistix "BLS4G3D1609DS1S00") = All good
16GB (2x8GB) (also ballistix "BLS8G3D1609DS1S00") = Segfault
24GB (2x8 + 2x4) all slots populated = Segfault
Memtest = No errors
Tried various stress-test apps, I was unable to get segfault, only with steam.
No test ever produced any errors, no matter how long I have ran it.
Steam/dota2 crashes in approx 15minutes 

However, I have started to think it might be related to bios/firmware that was written by someone with 2 left hands...

**Comment by madIlama** [comment]
> I try with two computers but they have the same result both opengl and vulkan.
> 
>     1. Laptop with Intel + Nvidia on Solus. (8 GB memory)
> 
>     2. PC with AMD Ryzen + AMDGPU on Fedora 31. (16 GB memory)
> 
> 
> ps. I suggest that you can try with the Roshpit Champion map.

I am crashing at some custom games too
I am using 16GB tho

After investigating a while , i found out, that its occur on some games that need "heavy" lifting
like Roshpit, IMBA, etc

In IMBA specifically, if you load test mode which is Demo mode (like normal Dota Demo/training) it will just load fine
because it just demo,a shorter map,with 1 hero to load with

But if you try to watch/queue 5v5 and 10v10 (which IMBA infamous for its lag on lower spec), boom, crash to desktop
I want to highlight again, that this happens just in this year.

Before in windows or linux, i never saw something like this

**Comment by Cabanur** [comment]
It turns out my crashes were my fault. At some point my motherboard settings were changed including some sort of automatic overclocking which seemed to be the cause of my system instability. I am sorry for wasting everyone's time on something that I should've checked before reporting a bug.

I am keeping this issue open since it seems like others have crashes that may need checking anyway.

**Comment by EarthDevour** [comment]
> It turns out my crashes were my fault. At some point my motherboard settings were changed including some sort of automatic overclocking which seemed to be the cause of my system instability. I am sorry for wasting everyone's time on something that I should've checked before reporting a bug.
> 
> I am keeping this issue open since it seems like others have crashes that may need checking anyway.

Can you share what was "overclocked" ? What MOBO you have ?

**Comment by Cabanur** [comment]
> Can you share what was "overclocked" ? What MOBO you have ?

I have an [Asus Prime X370-PRO](https://www.asus.com/Motherboards/PRIME-X370-PRO/), which has a feature where it'll try to do a light overclocking automatically with just a switch on the BIOS Settings. I can't remember the exact name for it though I can look it up if you really need me to. 

Disabling this feature inmediately made all the crashes go away.

**Comment by Cabanur** [comment]
Soon after posting here my issue was solved, unfortunately, the crashes came back. 

The crashes are not consistent. I have gotten crashes at any point in the game, while laning in early game or jungling in late game, with ho apparent reason. Most of the times the game simply crashes to desktop and I only have to restart dota. 

Other times, however, my screen will freeze frame o go to black. I am still able to hear and talk through TeamSpeak, but screen image is unchanging, even when trying to change TTY.

I assumed my GPU was at fault and replaced it with a brand new RX 5700XT. I had to get the drivers from Kernel.org because debian doesn't have them yet. I was able to get the game working but the crashes persist. 

I am attatching a kernel log for one of the crashes that seemed to output particularly interesting messages, like: 

```
May  8 17:10:27 sobremesa kernel: [ 3386.457666] pcieport 0000:00:03.1: AER: Multiple Corrected error received: 0000:00:00.0
May  8 17:10:27 sobremesa kernel: [ 3386.457679] pcieport 0000:00:03.1: AER: PCIe Bus Error: severity=Corrected, type=Data Link Layer, (Transmitter ID)
May  8 17:10:27 sobremesa kernel: [ 3386.457681] pcieport 0000:00:03.1: AER:   device [1022:1453] error status/mask=00001100/00006000
May  8 17:10:27 sobremesa kernel: [ 3386.457684] pcieport 0000:00:03.1: AER:    [ 8] Rollover              
May  8 17:10:27 sobremesa kernel: [ 3386.457686] pcieport 0000:00:03.1: AER:    [12] Timeout               
May  8 17:10:27 sobremesa kernel: [ 3386.461372] show_signal: 10 callbacks suppressed
May  8 17:10:27 sobremesa kernel: [ 3386.461374] traps: dota2[2042] general protection fault ip:7f75ff4d810d sp:7fffba05e2e0 error:0 in libclient.so[7f75fb66f000+44ee000]
May  8 17:10:28 sobremesa kernel: [ 3387.151984] pcieport 0000:00:03.1: AER: Multiple Corrected error received: 0000:00:00.0
May  8 17:10:28 sobremesa kernel: [ 3387.151998] pcieport 0000:00:03.1: AER: PCIe Bus Error: severity=Corrected, type=Data Link Layer, (Transmitter ID)
May  8 17:10:28 sobremesa kernel: [ 3387.152002] pcieport 0000:00:03.1: AER:   device [1022:1453] error status/mask=00001100/00006000
May  8 17:10:28 sobremesa kernel: [ 3387.152005] pcieport 0000:00:03.1: AER:    [ 8] Rollover              
May  8 17:10:28 sobremesa kernel: [ 3387.152007] pcieport 0000:00:03.1: AER:    [12] Timeout   
```

That seems to be the specific  crash. 

**I'm interested in knowing if anyone is able to tell me if this is a software or hardware issue. I understand this is not a tech support forum, but a simple "this has nothing to do with dota or steam software" would suffice. I would hate to replace the rest of the hardware and find I'm still having the same issue.**



I'm finding less performance than expected with a 5700XT. At 1440p, I'm only getting around 100-120 fps. The game will sometimes jump to 140+fps, and when this happens i'll hear the gpu fans ramping up, but the fans and the fps will wind back down for no apparent reason within a few seconds. This doesn't seem to correlate with more "stuff" happening on-screen.


[Here](https://github.com/ValveSoftware/Dota-2/files/4601554/Desktop.zip) are some dumps that i managed to get and a trimmed `kern.log`.  Thanks for your help.

**Comment by Cabanur** [comment]
I was out of options as to how to tell for sure if this was a software or hardware issue and eventually I thought of installing windows and trying. The game continues to crash and I still have ~100 fps on a RX 5700 XT so now I'm confident in saying it's definitely a hardware issue.
